Examples/BufferedExchange/src/MyWorker.hpp

```diff
@@ -0,0 +1,126 @@
+#pragma once
+#include "Arp/System/Commons/Threading/Thread.hpp"
+
+#include <queue>
+#include <utility>
+#include <chrono>
+
+namespace BufferedExchange
+{
+    class MyWorker
+    {
+
+    public: // Methods the Threads call.
+
+        //for automatic repeating WorkerThread
+        void RunSingle(void)
+        {
+            ++this->RunCount;
+            process();
+        };
+
+        //"Endless loop" for normal thread
+        void Run(void *)
+        {
+		while (!this->Stop)
+		{
+			RunSingle();
+			Arp::Thread::Sleep(1000);
+		}
+        };
+
+        //Static thread implementation
+        static void RunStatic(void *pParam)
+        {
+            ((MyWorker *)pParam)->Run(nullptr);
+        };
+
+    public: // Construction
+        MyWorker() : RunCount(0),
+                     Stop(true)
+        {
+            Log::Info("------------------- MyWorker Constructed: {0}",Stop);
+        }
+        ~MyWorker()
+        {
+            Log::Info("------------------- MyWorker Deconstructed: {0}",Stop);
+            Stop = true;
+        }
+    public: // Properties
+        int RunCount;
+        bool Stop;
+        Arp::Mutex d1;
+
+    private:
+        // use thread safe data storage of your choice or implement with semaphores|mutex etc.
+
+        // active queue for processing
+        std::queue<std::pair<double, int>> data_toprocess;
+
+        // active queue for storing
+        std::queue<std::pair<double, int>> data_store;
+
+        // Timepoint of last SetData command to log time between entries.
+        std::chrono::high_resolution_clock::time_point last_time{};
+
+    private:
+        void process()
+        {
+            try
+            {
+                while (!data_toprocess.empty() && !Stop)
+                {
+                    // Do something with the first element of this queue
+                    if (data_toprocess.front().second % (int)1000 == (int)0)
+                    {
+                        Log::Info("--- Thread time:{0} data:{1}", data_toprocess.front().first, data_toprocess.front().second);
+                    }
+                    data_toprocess.pop();
+                }
+
+                while (data_store.size() < 10 && !Stop)

```

Comment:
Dont Swap so often else these logs flood the Log.

```suggestion
                while (data_store.size() < 100 && !Stop)

```